blender-cli/src/deposit.rs

```diff
@@ -20,18 +22,27 @@ pub(super) fn do_deposit(
         ..
     } = cmd;
 
-    let pk = match fs::read(proving_key_file) {
-        Ok(bytes) => bytes,
-        Err(e) => panic!("Could not read pk: {}", e),
-    };
-
     let mut rng = rand::thread_rng();
 
     let trapdoor: Trapdoor = rng.gen::<u64>();
     let nullifier: Nullifier = rng.gen::<u64>();
     let note = compute_note(token_id, token_amount, trapdoor, nullifier);
 
     let circuit = DepositRelation::new(note, token_id, token_amount, trapdoor, nullifier);
+
+    let pk = match fs::read(proving_key_file) {
+        Ok(bytes) => bytes,
+        Err(_e) => {
+            let system = NonUniversalProvingSystem::Groth16;
+            let RawKeys { pk, vk } = system.generate_keys(circuit.clone());
+
+            fs::write("deposit.pk.bytes", pk.clone()).unwrap();
+            fs::write("deposit.vk.bytes", vk).unwrap();
```

Comment:
actually vk won't be of any use with this CLI

Reply:
it was to have it for registering in the pallet :)